src/Editor.elm

```diff
@@ -634,34 +610,16 @@ viewControls (( t, _ ) as z) =
                 in
                 if t.node.gui.controlsShown then
                     [ button [ class "button", onClick (ExpandUnary Alpha z) ] [ text "Add α" ]
-                    , div [ class "onclick-menu add", tabindex 0 ]
-                        [ ul [ class "onclick-menu-content" ]
-                            [ li [] [ button [ onClick (ExpandUnary Alpha z) ] [ text "Add α" ] ]
-                            , li [] [ button [ onClick (ExpandBinary Beta z) ] [ text "Add β" ] ]
-                            , li [] [ button [ onClick (ExpandUnaryWithSubst Gamma z) ] [ text "Add γ" ] ]
-                            , li [] [ button [ onClick (ExpandUnaryWithSubst Delta z) ] [ text "Add δ" ] ]
-                            , li [] [ button [ onClick (ExpandUnary Refl z) ] [ text "Add Reflexivity" ] ]
-                            , li [] [ button [ onClick (ExpandUnary Leibnitz z) ] [ text "Add Leibnitz" ] ]
-                            , li [] [ button [ onClick (ExpandUnary MP z) ] [ text "Add MP" ] ]
-                            , li [] [ button [ onClick (ExpandUnary MT z) ] [ text "Add MT" ] ]
-                            , li [] [ button [ onClick (ExpandBinary Cut z) ] [ text "Add Cut" ] ]
-                            , li [] [ button [ onClick (ExpandUnary HS z) ] [ text "Add HS" ] ]
-                            , li [] [ button [ onClick (ExpandUnary DS z) ] [ text "Add DS" ] ]
-                            , li [] [ button [ onClick (ExpandUnary NCS z) ] [ text "Add NCS" ] ]
-                            , li [] [ button [ onClick (ExpandBinary ECDF z) ] [ text "Add ECDF" ] ]
-                            , li [] [ button [ onClick (ExpandBinary ECDT z) ] [ text "Add ECDT" ] ]
-                            , li [] [ button [ onClick (ExpandUnary ESFF z) ] [ text "Add ESFF" ] ]
-                            , li [] [ button [ onClick (ExpandUnary ESFT z) ] [ text "Add ESFT" ] ]
-                            , li [] [ button [ onClick (ExpandUnary ESTF z) ] [ text "Add ESTF" ] ]
-                            , li [] [ button [ onClick (ExpandUnary ESTT z) ] [ text "Add ESTT" ] ]
-                            ]
-                        ]
-                    , div [ class "onclick-menu del", tabindex 0 ]
-                        [ ul [ class "onclick-menu-content" ]
+                    , ruleMenu ExpandUnary ExpandUnaryWithSubst ExpandBinary "Add ▾" "Add" "add" config z
+                    , div [ class "onclick-menu" ]
+                        [ button [ class "onclick-menu del", tabindex 0 ] [
+                            text "Delete ▾",
+                            ul [ class "onclick-menu-content" ]
```

Comment:
Perhaps there should be a `menu` function that would be used here as well as in `ruleMenu`.